In [36]:
import math

import pandas as pd
import scipy.stats as st

## HW 1
Chapter 12: 12.6, 12.8, 12.15, 12.19 (see HW 1 output)

### 12.6


In [4]:
# pop = 5th grade students who come from families which English is not primary language
# goal = determine if written English assessment scores differ between two different groups of children
#        who do and do not enter a new program

# g1 = 500
# g2 = 500
#
# vals = {
#     'student1': {
#         'prog': 'old prog',
#         'lang': 'not spoken'
#     },
#     'student2': {
#         'prog': 'old prog',
#         'lang': 'spoken'
#     },
#     'student3': {
#         'prog': 'new prog',
#         'lang': 'not spoken'
#     },
#     'student4': {
#         'prog': 'new prog',
#         'lang': 'spoken'
#     },
#
# }
#
# df = pd.DataFrame(vals).T
#
# n = g1 + g2
#
# df['prog'] = 1 if df['prog'] == "new prog" else 0
# df['lang'] = 1 if df['lang'] == "spoken" else 0

# S = b0 + (b1 * P) + (b2 * E) + (b3*P*E) + std_err

#### a
- b0 - b3 are mathematical constants, scalars
-- b0: default, not in program, english not spoken at home
-- b1: in program, english not spoken
-- b2: not in program, english spoken
-- b2: in program, english spoken
- b1 * P means it is true that the student was in the new program - this will help determining the programs' impact on written scores
- b2 * E means it is true that the student grew up in a home where English was spoken - this will help tell us whether or not English being spoken at home had an impact on their written scores
- b3 * P * E means it is true that the student was in the new program AND grew up in a home where English was spoken - this intersection is important as these students may test highest.
- Standard error is given, or assumed 0 if variance is normally distributed


#### b

The mean in diff scores when looking at students who don't speak at home, we will notice in our model that b2, our flag to determine whether English is spoken (1) vs. not spoken (0).

When looking to observe mean scores among these students, we will see scores change in the relative degree as determined by b2 if english was spoken.

#### c

The mean in diff scores when looking at students among old and new programs, we will notice in our model that 12, is our flag to determine whether they were in the new (1) or old (0) program.

When looking to observe mean scores among these students, we will see scores change in the relative degree as determined by b1 if a student was in the program.

#### d

Why is b3 * P * E necessary?

We need to understand the intersection(s) of variables to understand their correlation. It is important to understand how the impact of having been in the new program AND whether english was spoken at home to see how it impacts the model. One can hypothesize that students who speak english at home scored different from each other between the programs.

### 12.8

In [5]:
# S4 = score at end of 4th grade
# S5 = score at end of 5th grade
# D = S5 - S4

# model_1 = D = b0 + (b1 * P) + (b2 * E) + (b3 * G) + (b4*P*E) + (b5*P*G) + (b6*E*G) + std_err
# model_2 = S5 = b0 + (b1 * P) + (b2 * E) + (b3 * G) + (b4*P*E) + (b5*P*G) + (b6*E*G) + (b7*S4) + std_err

#### a

In [6]:
# D = S5 - S4 = b0 + (b1 * P) + (b2 * E) + (b3 * G) + (b4*P*E) + (b5*P*G) + (b6*E*G) + std_err
# S5 = b0 + (b1 * P) + (b2 * E) + (b3 * G) + (b4*P*E) + (b5*P*G) + (b6*E*G) + std_err + S4

#### b
Model 2 is better as it applies a relative scale to the students' grade from the 4th grade. If a student scores highly in the prior year without a scalar, it will heavily skew how our model predicts our students' scores

### 12.15

Regression Analysis: Revenue versus Distance, Population

Analysis of Variance

    Source          DF      Adj SS      Adj MS      F-Value     P-Value
    Regression      2	35286       17643       14.65	0.000
    Distance	    1	1494	1494	1.24	0.279
    Population	    1	35243	35243	29.27	0.000
    Error	    19	22874	1204
    Total	    21	58160


Model  Summary

    S   R-sq	R-sq(adj)	R-sq(pred}
    34.6974	60.67%	56.53%  19.37%

Coefficients

    Term	    Coef	SE Coef	T-Value	P-Value	VIF
    Constant	144.4	39.5	3.65	0.002
    Distance	0.169	0.152	1.11	0.279	1.06
    Population	1.094	0.202	5.41	0.000	1.06

Regression Equation
Revenue = 144.4 + 0.169 Distance + 1.094 Population

    Residual Error	56	5.304	0.095
    Total		    59	68.500

    Source	    DF	Seq SS
    Additive	1	58.893
    AdditiveA2	1	4.202
    CuLevel	    1	0.101


Unusual Observations

    Obs	Additive
    52	100

    FeedEffR 3.8700

    Fit
    4.5413

    SE   Fit 0 .0967

    Residual
    -0 . 6713

    St  Resid
    -2.30R


    R denotes an observation with a large standardized residual.



#### a

Ho: No overall predictive value
Ha: There is an overall predictive value

alpha = 0.01

When looking at our ANOVA output, along our Regression row - we can note our F-value of 14.65, returning a p-value of 0.000. Since 0.000 < 0.01, we can reject the claim there is no predictive value and state there is enough evidence to conclude our model predicts revenue.

#### b

Ho: Distance to the hub is not significant
Ha: Distance to the hub is

alpha = 0.05

Referring to our output, we can note our F-value of 1.24, returning a p-value of 0.216. Since 0.216 > 0.05, we can conclude the distance to the hub is not a significant predictor to predict revenue

#### c - 95% confidence interval

In [45]:
b_dist = 0.169
se_dist = 0.152

df_err = 19

t = abs(st.t.ppf(0.05 / 2, df_err))
me_dist = t * se_dist

lower = b_dist - me_dist
upper = b_dist + me_dist
print(f"95% confidence interval: ({round(lower, 4)}, {round(upper, 4)})")

95% confidence interval: (-0.1491, 0.4871)


We can note that we are 95% confident the amount distance impacts the predicted value ranges from -.1491 and .4871. Through this range, we can see why distance does not provide value to our model's predicted revenue.

#### d - hypothesis testing

b2 = slope associated with population

Ho: b2 = 0.5
Ha: b2 > 0.5

alpha = 0.05
n = 22

In [50]:
b_pop = 1.094
b2 = 0.5
se_pop = 0.202

t = (b_pop - b2) / se_pop
n = 22
df = n - 1
print(f"Test statistic: {round(t, 4)}")

alpha = 0.05
crit_val = abs(st.t.ppf(alpha, df))
print(f"Critical value: {round(crit_val, 4)}")

p_val = st.t.cdf(t, df)
print(f"P value: {round(1 - p_val, 4)}")

Test statistic: 2.9406
Critical value: 1.7207
P value: 0.0039


When observing the slope of our linear regression model, we can state we are 95%, and even 99% confident that our slope associated with population is greater than 0.5.


 ### 12.19

Regression Analysis: Systolic versus Age, Weight

Analysis of Variance

    Source      DF          Adj  SS         Adj MS        F-Value     P-Value
    Regression   4          1551.66         387.915         75.81      0.000
    Age	1	35.12	35.124	6.86	0.016
    Weight	1	20.40	20.400	3.99	0.060
    Age*Age	1	8.41	8.411	1.64	0.214
    Weight*Weight	1	13.83	13.834	2.70	0.116
    Error	20	102.34	5.117
    Lack-of-Fit	16	85.84	5.365	1.30	0.439
    Pure Error	4	16.50	4.125
    Total	24	1654.00


Model Summary

    s	R-sq	R-sq ( adj )	R-sq(pred)
    2.26210	93 . 81%	92.57%	89.21%

Coefficients

    Term        Coef        SE Coef         T-Value         P-Value     VIF
    Constant    9.6	20.8	0.46	0.649
    Age	11.58	4.42	2.62	0.016	122.70
    Weight	23.6	11.B	2.00	0.060	150.23
    Age*Age	-0.649	0.506	-1.28	0.214	133.09
    Weight*Weight	-2.84	1.72	-1.64	0.116	156.45

Regression Equation
Systolic = 9.6 + 11.58 Age + 23.6 Weight - 0.649 Age*Age - 2.84 Weight*Weight

Fits and Diagnostics for Unusual Observations

    Obs
    8
    16

    Systolic
    110.00
    103.00

    Fit
    104.82
    104.45

    Resid
    5.18
    -1.45

    Std Resid
    2 . 46 R
    -1.33	x


    R Large residual X  Unusual X


    NOTE:  SS Regression for reduced model (without quadratic terms) is 1521.53


#### a - model analysis summary

Overall, we have a mulitple linear regression model that is significant enough to predict our response variable, systolic blood pressure. Our independent variables are significant at alpha = 0.10, but weight does not pass when alpha = 0.05. This isn't the end of the world, but something to keep in mind.

We have a nice adjusted r-square value, displaying an evident 92.57% explanation in variation in data, which is really nice. We can finally note we have squared values for our independent variables - a new thing for me!

#### b - second order hypothesis testing

Observing our p-values for our second order values, we can see that they both exceed alpha = 0.05 and 0.10 respectively. This is important as it shows these values are not significant when predicting systolic blood pressure. This makes sense since these are the same values by relative scale.

#### c

When looking at our models, between first and second order, it makes the most sense to go with our first order model. While our second order model gives us a nice response, that doesn't mean it's the most optimal. We are not calcluating anything "new" per se, as we square these values.

In [54]:
# !jupyter nbconvert --to html HW1.ipynb --quiet